bluepysnap/edges/edge_population.py

```diff
@@ -576,3 +576,19 @@ def iter_connections(
     def h5_filepath(self):
         """Get the H5 edges file associated with population."""
         return self.population_config["edges_file"]
+
+    @cached_property
+    def spatial_synapse_index(self):
+        """Access to edges spatial index."""
+        try:
+            from spatial_index import open_index
+        except ImportError as e:
+            raise BluepySnapError(
+                (
+                    "Spatial index is for now only available internally to BBP. "
+                    "It requires `spatial_index`, an internal package."
+                )
+            ) from e
+
+        properties = self._circuit.to_libsonata.edge_population_properties(self.name)
+        return open_index(properties.spatial_synapse_index_dir)
```

Comment:
Accessing the property `spatial_synapse_index` or `spatial_segment_index` in a circuit without indexes would cause the error:
```
RuntimeError: filesystem error: cannot make canonical path: Invalid argument []
```
A more descriptive message would be more useful IMHO.

Reply:
yup, agreed

Reply:
fixed